cmake/external/featurizers.cmake

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# This source code should not depend on the onnxruntime and may be built independently
+
+set(featurizers_URL "https://github.com/microsoft/FeaturizersLibrary.git")
+set(featurizers_TAG "006df6bff45dac59d378609fe85f6736a901ee93")
+
+set(featurizers_pref FeaturizersLibrary)
+set(featurizers_ROOT ${PROJECT_SOURCE_DIR}/external/${featurizers_pref})
+set(featurizers_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/external/${featurizers_pref})
+
+# Only due to GIT_CONFIG
+# Uncoment UPDATE_COMMAND if you work locally
+# on the featurizers so cmake does not undo your changes.
+if (WIN32)
+    ExternalProject_Add(featurizers_lib
+            PREFIX ${featurizers_pref}
+            GIT_REPOSITORY ${featurizers_URL}
+            GIT_TAG ${ngraph_TAG}
+            # Need this to properly checkout crlf
+            GIT_CONFIG core.autocrlf=input
+            SOURCE_DIR ${featurizers_ROOT}
+            # Location of CMakeLists.txt
+            SOURCE_SUBDIR src/Featurizers
+            BINARY_DIR ${featurizers_BINARY_DIR}
+#            UPDATE_COMMAND ""
+            INSTALL_COMMAND ""
+        )
+else()
+    ExternalProject_Add(featurizers_lib
+            PREFIX ${featurizers_pref}
+            GIT_REPOSITORY ${featurizers_URL}
+            GIT_TAG ${featurizers_TAG}
+            SOURCE_DIR ${featurizers_ROOT}
+            # Location of CMakeLists.txt
+            SOURCE_SUBDIR src/Featurizers
+            BINARY_DIR ${featurizers_BINARY_DIR}
+            CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+#            UPDATE_COMMAND ""
+            INSTALL_COMMAND ""
+        )
+endif()
+
```

Comment:
Just wondering, does the lib has any C++ types in its public interface?

Reply:
Well, they are not public  outisde ORT. ORT consumes this library.

---
In reply to: [357838315](https://github.com/microsoft/onnxruntime/pull/2643#discussion_r357838315) [](ancestors = 357838315)